etc/docker-stacks/all-spark-notebook/Dockerfile

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) Jupyter Development Team.
+FROM jupyter/all-spark-notebook:dbb9c7b50531
+
+MAINTAINER Jupyter Project <jupyter@googlegroups.com>
+
+USER jovyan
+
+# Install Kernel Gateway
+RUN pip install jupyter_kernel_gateway==0.1.0
+
+# Configure container startup
+ENTRYPOINT ["jupyter", "kernelgateway"]
```

Comment:
The base image uses tini as the entrypoint which we need to keep because tini makes sure that orphaned child processes are adopted properly whereas jupyter does not. 

By default, I think we can do:

```
ENTRYPOINT ["tini", "--", "jupyter", "kernelgateway"]
CMD ["--KernelGatwayApp.ip", "0.0.0.0"]
```

But then users can override it on run like:
```
docker run -d -P jupyter-incubator/kernel_gateway --KernelGatewayApp.seed_notebook=foo.ipynb # etc.
```